### PR TITLE
added gitignore file

### PR DESCRIPTION
**Pull request by Craigryy**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-      - pgdata:/var/lib/postgresql/data
-
-volumes:
-  pgdata: {}
-    
+version: '3'
+
+services:
+  flask_app:
+<<<<<<< HEAD
```
Comment (andela-sjames) [inline docker-compose.yml]: This ```<<<<<<< HEAD``` should not be here.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+    build: .
+=======
```
Comment (andela-sjames) [inline docker-compose.yml]: should not be here

File: docker-compose.yml [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
+    build:
+      context: ./flask_app
+      dockerfile: Dockerfile
+>>>>>>> f923958493f9c3a30ec5a8d3f6ff892b4a0500b9
```
Comment (andela-sjames) [inline docker-compose.yml]: should not be here

File: flask_app/authenticate.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+        return f(current_user, *args, **kwargs)
+
+    return decorated
+
+# create a user
+
+# create an endpoint to get all users
+
+
```
Comment (andela-sjames) [inline flask_app/authenticate.py]: comment and function far apart

File: config.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+from flask_sqlalchemy import SQLAlchemy
+from flask import Flask
+
+# Create the Flask application
+app = Flask(__name__)
+
+# Configure the database
+app.config['SQLALCHEMY_DATABASE_URI'] = f'postgresql://{POSTGRES_USER}:{POSTGRES_PASSWORD}@{POSTGRES_HOST}/{POSTGRES_DB}'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['SECRET_KEY'] = 'jesusislord'
```
Comment (andela-sjames) [inline config.py]:
```python
app.config['SECRET_KEY'] = os.environ.get('SECRET_KEY',  'jesusislord')
```

secrets should be from env value

Comment (andela-sjames) [reply config.py]: Same to all config

File: docker-compose.yml [hunk 6]
```diff
@@ -1,28 +1,23 @@
...
-      - POSTGRES_PASSWORD=Favour98
-      - POSTGRES_USER=postgres
-      - POSTGRES_DB=kittie
-    volumes:
-      - pgdata:/var/lib/postgresql/data
-
-volumes:
-  pgdata: {}
-    
+version: '3'
```
Comment (andela-sjames) [inline docker-compose.yml]: why the drop from `3.9` to `3`

File: config.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+port = 5000
```
Comment (andela-sjames) [inline config.py]: Can we have the port number centralized and pulled from the env file

File: flask_app/authenticate.py [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+@auth_app.route('/test', methods=['GET'])
+def test():
+    return make_response(jsonify({'message': 'test route'}), 200)
+
+
+# create a token_required decorator which requires a header "x-access-token" and a token .
+def token_required(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        secret_key = current_app.config['SECRET_KEY']
```
Comment (andela-sjames) [inline flask_app/authenticate.py]: Defined but not used ?

File: docker-compose.yml [hunk 9]
```diff
@@ -1,28 +1,23 @@
...
+
```
Comment (andela-sjames) [inline docker-compose.yml:5]: You should define your volume storage local path before mounting and using it like this
```yaml
volumes:
  postgres_data: {}


volumes:
      - postgres_data:/var/lib/postgresql/data
```

That way it works on any machine like Linux, MacOS and windows

File: flask_app/bookmodel.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        return make_response(('book with id:{} was not found' .format(book_id)))
+
+    book = BookModel.query.get(id)
+    if book:
+        book.title = request.json.get('title', book.title)
+        book.author = request.json.get('author', book.author)
+        book.category_id = request.json.get('category_id', book.category_id)
+        book.category_id = category.id
+        book.save()
+        return make_response(jsonify({"message": "Book updated successfully", "book": book.to_json()}), 200)
```
Comment (andela-sjames) [inline flask_app/bookmodel.py:57]: Check line length.

File: flask_app/bookmodel.py [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+    return make_response(jsonify({"message": "Book not found"}), 404)
+
+
+# Write a function named `delete_book` which updates an existing book using `DELETE` method,
```
Comment (andela-sjames) [inline flask_app/bookmodel.py]: comment length

File: docker-compose.yml (hunk 9 above)
Comment (andela-sjames) [reply docker-compose.yml:5]: Here is a sample to reference: https://github.com/andela-sjames/jotpad/blob/main/docker-compose.yml

**Review by andela-sjames — COMMENTED** [review]
Let's start with this fix before we continue

File: flask_app/authenticate.py (hunk 8 above)
Comment (andela-sjames) [reply flask_app/authenticate.py]: `secret_key` defined but not used ?

File: flask_app/bookmodel.py (hunk 11 above)
Comment (Craigryy) [reply flask_app/bookmodel.py]: done

File: docker-compose.yml (hunk 9 above)
Comment (Craigryy) [inline docker-compose.yml:5]: done

File: flask_app/authenticate.py (hunk 8 above)
Comment (Craigryy) [reply flask_app/authenticate.py]: corrected

File: docker-compose.yml [hunk 12]
```diff
@@ -1,28 +1,26 @@
...
+    build: .
+    ports:
+      - 5000:5000
+    volumes:
+      - ./flask_app:/app/flask_app
+    depends_on:
+      - db
+    environment:
+      - FLASK_ENV=development
+      - DATABASE_URL=postgresql://postgres:Favour98@localhost/halo
```
Comment (andela-sjames) [inline docker-compose.yml]: This is not used by the application; this is not the source of the truth.

File: config.py [hunk 13]
```diff
@@ -0,0 +1,27 @@
+import os
+
+POSTGRES_HOST = os.environ.get('POSTGRES_HOST', 'localhost')
```
Comment (andela-sjames) [inline config.py]: `POSTGRES_HOST` and the others should be added to `docker-compose.yml`

Comment (andela-sjames) [reply config.py]: Make migrations and test everything on docker-compose

File: docker-compose.yml (hunk 12 above)
Comment (Craigryy) [reply docker-compose.yml]: done

**Review by andela-sjames — APPROVED** [review]
I will test the full functionality and check the code style again in the next PR

**Pull request merged**